src/main/java/domain/GamePlayer.java

```diff
@@ -0,0 +1,88 @@
+package domain;
+
+import io.Printer;
+import io.Receiver;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GamePlayer {
+    Printer printer;
+    Receiver receiver;
+    Generator generator;
```

Comment:
코드 전반에 걸쳐 인스턴스 변수들의 접근 제어자가 package-private으로 설정되어 있습니다.
접근 제어자를 특별히 지정하지 않으면 package-private으로 설정되는데 이는 같은 패키지 내의 다른 클래스에서 접근이 가능한 제어자 입니다.

객체는 자신의 상태를 자신만이 변경할 수 있도록 자신의 상태는 철저히 숨겨야 합니다. [[객체의 자율성 참고]](https://medium.com/@hongseongho/%EC%9E%90%EC%9C%A8%EC%A0%81%EC%9D%B8-%EA%B0%9D%EC%B2%B4-%EA%B3%B5%EB%8F%99%EC%B2%B4-%EC%84%A4%EA%B3%84%ED%95%98%EA%B8%B0-2c23be971a73)
접근 제어자는 특별한 경우가 아니라면 private으로 설정하는 것이 안전하지 않을까요?

```suggestion
    private final Printer printer;
    private final Receiver receiver;
    private final Generator generator;
```
또한 Printer나 Receiver, Generator 모두 한번 생성하면 다른 객체를 따로 생성할 일이 없으니 final로 제한해도 좋을 것 같습니다. 😄